R/MzTabReader.R

```diff
@@ -389,8 +432,8 @@ getMSMSScans = function(identified_only = FALSE)
   ## e.g. spectra.ref might be 'ms_run[1]:controllerType=0 controllerNumber=1 scan=13999'
   ##                        or 'ms_run[2]:spectrum=33'
   ##      --> extract scan as numeric, since string compare is insufficient for numbers ("13999" > "140")
-  res$scan = as.numeric(gsub(".*scan=(\\d*)[^\\d]*|.*spectrum=(\\d*)[^\\d]*", "\\1\\2", res$spectra.ref))
-  stopifnot(all(!is.na(res$scan)))
+  res$scan = as.numeric(gsub(".*index=(\\d*)[^\\d]*|.*scan=(\\d*)[^\\d]*|.*spectrum=(\\d*)[^\\d]*", "\\1\\2", res$spectra.ref))
```

Comment:
should be "\\1\\2\\3"
and can you test if `".*index=(\\d*)|.*scan=(\\d*)|.*spectrum=(\\d*)"` also works?